sys/usb/usbus/cdc/acm/cdc_acm.c

```diff
@@ -126,7 +126,20 @@ static size_t _gen_full_acm_descriptor(usbus_t *usbus, void *arg)
 /* Submit (ACM interface in) */
 size_t usbus_cdc_acm_submit(usbus_cdcacm_device_t *cdcacm, const uint8_t *buf, size_t len)
 {
-    return tsrb_add(&cdcacm->tsrb, buf, len);
+    if (cdcacm->state != USBUS_CDC_ACM_LINE_STATE_DISCONNECTED) {
+        return tsrb_add(&cdcacm->tsrb, buf, len);
+    }
+    // stuff as much data as possible into tsrb, discarding the oldest
+    size_t n = tsrb_free(&cdcacm->tsrb);
+    if (len > n) {
+        n += tsrb_drop(&cdcacm->tsrb, len - n);
```

Comment:
Unfortunately, this is not thread safe, the tsrb module is only thread safe when a single producer and a single consumer is used (which also means that the current situation is also not thread safe).

Easiest solution would be to disable interrupts during any tsrb operations or guard them with a mutex.

Reply:
[See also this documentation](https://github.com/RIOT-OS/RIOT/blob/master/sys/include/tsrb.h#L15-L16)

Reply:
Thanks, fixed that. Also squashed a few commits. Only thing still needing improvement is the test output, then.